### PR TITLE
Move Incidents from LegStep to RouteLeg

### PR DESCRIPTION
**Pull request by korshaknn**
A regular json from directions API looks like
```
{
  "routes": [
    {
      "legs": [
        {
          "steps": [],
          "incidents": [],
          ...
        }
      ],
      ...
    }
  ],
  "waypoints": [],
  ...
}
```
`Incidents` should be moved from `LegStep` to `RouteLeg`

### CODE REVIEW

**Comment by Guardiola31337** [comment]
Did we test that Directions requests return incidents and if they're parsed properly when using the `DirectionsService`? It'd be also great to test with a `SNAPSHOT` from the Navigation SDK and if it works release a beta so we can integrate it.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java [hunk 1]
```diff
@@ -85,6 +85,14 @@ public static Builder builder() {
   @Nullable
   public abstract List<LegStep> steps();
 
+  /**
+   * A list of incidents that occur on the route.
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java:89]:
```suggestion
   * A list of incidents that occur on this leg.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java [hunk 2]
```diff
@@ -199,6 +207,14 @@ public static RouteLeg fromJson(String json) {
      */
     public abstract Builder steps(@Nullable List<LegStep> steps);
 
+    /**
+     * A list of incidents that occur on the route.
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java:211]:
```suggestion
     * A list of incidents that occur on this leg.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java (hunk 1 above)
Comment (korshaknn) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java:89]: fixed

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java (hunk 2 above)
Comment (korshaknn) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteLeg.java]: fixed

**Comment by korshaknn** [comment]
@Guardiola31337 have added an `Incidents` parsing test with `MockWebServer`.

**Comment by korshaknn** [comment]
@Guardiola31337 ready

Screen Shot 2020-10-20 at 16 02 41

**Review by Guardiola31337 — APPROVED** [review]
:shipit:

**Pull request merged**